### PR TITLE
Fix UID and GID when providing IDs instead of names

### PR DESCRIPTION
**Pull request by paulRbr**
__Hi there!__

I stumbled upon this problem wanting to set my `god` task as a user given it's uid instead of it's name.

There might be a smarter way to do this, but this PR handles the case.

### CODE REVIEW

**Comment by paulRbr** [comment]
Hi @eric, I've rebased this PR. Any chance to get this into `master` soon?

File: lib/god/socket.rb [hunk 1]
```diff
@@ -95,8 +95,14 @@ def start
       end
 
       if File.exists?(self.socket_file)
-        uid = Etc.getpwnam(@user).uid if @user
-        gid = Etc.getgrnam(@group).gid if @group
+        if @user
+          user_method = @user.is_a?(Integer) ? :getpwuid : :getpwnam
+          uid = Etc.send(user_method, @user).uid
```
Comment (eric) [inline lib/god/socket.rb]: It seems like this should also do the same gid defaulting logic from the `Process` class too.

**Comment by paulRbr** [comment]
Thanks for the review @eric! I've updated the PR.

**Comment by eric** [comment]
Looks good!

**Pull request merged**